uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Helper/Menu/Parser.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+use CRM_HRCore_Menu_Navigation as Navigation;
+
+class CRM_HRCore_Helper_Menu_Parser {
+
+  /**
+   * Builds the navigation menu tree and returns in a format expected by Civi
+   * in order to generate the navigation menu.
+   *
+   * @param Navigation $navigationMenu
+   *
+   * @return array
+   *  Sample return format:
+   *  [
+   *    'attributes' =>
+   *      [
+   *        'label' => 'Components',
+   *         'url' => 'civicrm/bla',
+   *         'permission' => 'sample permission',
+   *         'operator' => 'AND',
+   *      ],
+   *   ],
+   */
+  public static function getNavigationTree(Navigation $navigationMenu) {
+    return self::buildMenuTree($navigationMenu->getItems());
+  }
+
+  /**
+   * Builds the navigation menu tree for the menu items for the given parent ID.
+   *
+   * @param array $items
+   * @param int $parentID
+   *
+   * @return array
+   */
+  private static function buildMenuTree(array $items, $parentID = 0) {
+    STATIC $navID = 1;
```

Comment:
How about using an extra argument in the `buildMenuTree` to accept current highest ID? 

I'm just saying that because if you (for some reason) had two calls to `getNavigationTree` then the $navID would not be reset. See [SO](https://stackoverflow.com/a/27879260/1196369)

Reply:
Good point. Will look into that.